util/helm/cmd.go

```diff
@@ -219,6 +219,9 @@ func (c *Cmd) Fetch(repo, chartName, version, destination string, creds Creds) (
 	if creds.Password != "" {
 		args = append(args, "--password", creds.Password)
 	}
+	if creds.InsecureSkipVerify {
```

Comment:
Please change to `creds.InsecureSkipVerify && c.insecureSkipVerifySupported`, simiar to:

https://github.com/argoproj/argo-cd/blob/6f959506ab8acb1d220ca0b9f0e936fddbb6b277/util/helm/cmd.go#L160

The `--insecure-skip-tls-verify` was added recently to helm client and not supported by older versions.

Reply:
This might be needed for the login command as well?

Reply:
Sorry @elucidator, I did not notice your update. For consistency is good to add the same check to `Login` but not must have: `Login` command is only available in helm3 which supports skip verify so we can leave it as is.